src/urllib3/connection.py

```diff
@@ -362,6 +362,18 @@ def connect(self):
                 ssl_version=resolve_ssl_version(self.ssl_version),
                 cert_reqs=resolve_cert_reqs(self.cert_reqs),
             )
+            # In some cases, we want to verify hostnames ourselves
+            if (
```

Comment:
Love how this codifies all the situations where we disable `check_hostname` so nicely :star_struck: